modules/generic-extras/src/main/scala/io/circe/generic/extras/decoding/ConfiguredDecoder.scala

```diff
@@ -54,9 +56,12 @@ final object ConfiguredDecoder extends IncompleteConfiguredDecoders {
     decodeR: Lazy[ReprDecoder[R]],
     config: Configuration
   ) extends ConfiguredDecoder[A] {
+
+    private[this] val constructorNameCache: TrieMap[String, String] = new TrieMap()
+
     final def apply(c: HCursor): Decoder.Result[A] = decodeR.value.configuredDecode(c)(
       Predef.identity,
-      config.transformConstructorNames,
+      v => constructorNameCache.getOrElseUpdate(v, config.transformConstructorNames(v)),
```

Comment:
I'm wondering whether we should construct this function outside of `apply` to avoid allocations.

Reply:
I think, you are right. I'll extract it outside of apply.